src/Homepage/Posts/Posts.js

```diff
@@ -1,19 +1,41 @@
 import React from 'react'
 import ThumbUpIcon from '@mui/icons-material/ThumbUp';
-import SendIcon from '@mui/icons-material/Send';
+import DeleteIcon from '@mui/icons-material/Delete';
 import ChatIcon from '@mui/icons-material/Chat';
-import ShareIcon from '@mui/icons-material/Share';
-import InputOptions from '../InputOptions/InputOptions';
+import EditIcon from '@mui/icons-material/Edit';
 import Avatar from '@mui/material/Avatar';
 import "./Posts.css"
 
-function Posts({ name , description , message }) {
+import {
+  doc,
+  deleteDoc,
+  setDoc
+} from "firebase/firestore";
+
+import db  from "../../firebase"
+
+function Posts({value, name , description , message }) {
+
+  const handleDelete = async (idle) => {
+    const docRef = doc(db, "posts", idle);
+    await deleteDoc(docRef);
+  };
+
+  
+  const handleEdit = async (id) => {
+  const message = prompt("Enter post mesage") ;
```

Comment:
use controlled input. because if user types nothing in prompt and press OK or just press cancel button. The post content will be overwrite by empty string.

Reply:
done the changes thanks for reviewing